examples/verifying_mfd.py

```diff
@@ -1,16 +1,183 @@
 import desbordante
+import pandas as pd
+from operator import attrgetter
 
-TABLE = 'examples/datasets/theatres_mfd.csv'
-METRIC = 'euclidean'
-LHS_INDICES = [0]
-RHS_INDICES = [2]
-PARAMETER = 5
-
-algo = desbordante.mfd_verification.algorithms.Default()
-algo.load_data(table=(TABLE, ',', True))
-algo.execute(lhs_indices=LHS_INDICES, metric=METRIC,
-             parameter=PARAMETER, rhs_indices=RHS_INDICES)
-if algo.mfd_holds():
-    print('MFD holds')
-else:
-    print('MFD does not hold')
+GREEN_CODE = "\033[1;42m"
+RED_CODE = "\033[1;41m"
+BLUE_CODE = "\033[1;46m"
+DEFAULT_COLOR_CODE = "\033[1;49m"
+
+def print_results(verifier, data, parameter):
+    highlights_list = verifier.get_highlights()
+    if verifier.mfd_holds():
+        print(GREEN_CODE, "MFD holds", DEFAULT_COLOR_CODE)
+    else:
+        print(RED_CODE, "MFD does not hold due to the following items:",
+              DEFAULT_COLOR_CODE)
+        for ind, cluster in enumerate(highlights_list, start=1):
+            print(f"{BLUE_CODE} #{ind} cluster: {DEFAULT_COLOR_CODE}")
+            listPandas = list(map(lambda x: x.data_index, list(cluster)))
+            print(data.iloc[listPandas])
+            max_distance = max(cluster, key=attrgetter("max_distance")
+                               ).max_distance
+            print(f"Max distance: {max_distance} > {parameter}")
+        print()
+
+# Method that showcases distance between coordinates
+def mfd_L2(table="examples/datasets/addresses_coordinates.csv"):
+    METRIC = "euclidean"
+    METRIC_ALGORITHM = "calipers"
+    LHS_INDICES = [1]
+    RHS_INDICES = [2,3]
+    PARAMETERS = [1, 0.1, 0.01, 0.001, 0.0001]
+
+    print("To do that, let's explore the addresses_coordinates.csv and try "
+          "to verify the metric functional dependency in it.\n")
+
+    data = pd.read_csv(table, header=[0])
+    print(data)
+
+    for value in PARAMETERS:
+        algo = desbordante.mfd_verification.algorithms.Default()
+        algo.load_data(table=(table, ",", True))
+        algo.execute(lhs_indices=LHS_INDICES,
+            metric=METRIC,
+            metric_algorithm=METRIC_ALGORITHM,
+            parameter=value,
+            rhs_indices=RHS_INDICES)
+        print("Checking whether [Address] -> [Latitude, Longitude] AFD holds"
+              f" with parameter δ = {value}")
+        print_results(algo, data, value)
+
+    print("As we can notice from the results, lessening the value of a "
+          " parameter strictens the constraint, and, as such, yields more "
+          "violations the smaller the parameter. Namely, 0.001 is the first "
+          " value we tried that resulted in MFD no longer holding. The "
+          " algorithm also provides us with the culprits responsible for "
+          "MFD not holding, which are divided in "
+          f"{BLUE_CODE}clusters{DEFAULT_COLOR_CODE}.\n")
```

Comment:
I'm not entirely aware what the algorithm does, but.
* "culprits" - find a better word. Record pairs? Records? "that prevent"
* divided - is it actually "grouped into"?
```suggestion
          " value we tried that resulted in the MFD no longer holding. The "
          " algorithm also provides us with the records(?) that prevent "
          "the MFD from holding, which are grouped into "
          f"{BLUE_CODE}clusters{DEFAULT_COLOR_CODE}.\n")
```

Reply:
Changed to "The algorithm also provides us with record pairs that prevent MFD from holding, with those records being grouped into clusters that share a common left-hand side"